File: M10702130_HW1/JAVA/src/M10702130_HW1.java
import java.io.FileReader;
import java.io.IOException;
import java.util.Scanner;

public class M10702130_HW1 {

	public static void main(String[] args) {
		try {
			int start, end, ans, guess;
			
			Scanner keyIn  = new Scanner(System.in);
			Scanner fileIn = new Scanner(new FileReader("input.txt"));
			
			start = fileIn.nextInt();
			end   = fileIn.nextInt();
			ans   = fileIn.nextInt();
			
			while (true) {
				System.out.println("Please guess an integer number between " + start + " and " + end);
				
				guess = keyIn.nextInt();
				
				if (guess <= start || guess >= end) {
					System.out.println("The number you entered is not within the valid range.");
					System.out.println("Please try again!\n");
				} else if (guess < ans) {
					start = guess;
				} else if (guess > ans) {
					end = guess;
				} else {
					System.out.println("Bingo! You hit the bomb.");
					break;
				}
			}
			
			keyIn.close();
			fileIn.close();
		} catch (IOException e) {
			System.out.println(e.getMessage());
		}
	}
}

File: M10702130_HW1/CPP/M10702130_HW1/M10702130_HW1.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main()
{
	int start, end, ans, guess;
	ifstream fileIn;

	fileIn.open("input.txt");
	fileIn >> start >> end >> ans;

	while (true) {
		cout << "Please guess an integer number between " << start << " and " << end << endl;
		
		cin >> guess;

		if (guess <= start || guess >= end) {
			cout << "The number you entered is not within the valid range." << endl;
			cout << "Please try again!" << endl << endl;
		} else if (guess < ans) {
			start = guess;
		} else if (guess > ans) {
			end = guess;
		} else {
			cout << "Bingo! You hit the bomb." << endl;
			break;
		}
	}

	fileIn.close();

	return 0;
}